File: src/main/java/co/herod/adbwrapper/util/FileUtil.java
package co.herod.adbwrapper.util;

import org.jetbrains.annotations.NotNull;

import java.io.File;

public class FileUtil {

    public static long getAgeMillis(final File file) {
        return System.currentTimeMillis() - (file.exists() ? file.lastModified() : 0);
    }

    @NotNull
    public static File getFile(final String pathname) {
        final File file = new File(pathname);

        if (file.exists()) {
            return file;
        }

        final File folder = new File(getFolderPath(pathname));
        final boolean g = folder.exists() || folder.mkdirs();

        return file;
    }

    @NotNull
    private static String getFolderPath(final String pathname) {

        if (!pathname.contains("/") && pathname.contains(".")) {
            return "./";
        }

        try {
            final int lastIndexOf = pathname.lastIndexOf("/");
            return pathname.substring(0, lastIndexOf);
        } catch (final IndexOutOfBoundsException e) {
            return pathname;
        }
    }
}

File: src/main/java/co/herod/adbwrapper/util/FileUtil.kt
package co.herod.adbwrapper.util

import java.io.File

object FileUtil {

    fun getAgeMillis(file: File): Long {
        return System.currentTimeMillis() - if (file.exists()) file.lastModified() else 0
    }

    fun getFile(pathname: String): File {

        val file = File(pathname)

        if (file.exists()) {
            return file
        }

        val folder = File(getFolderPath(pathname))
        val g = folder.exists() || folder.mkdirs()

        return file
    }

    private fun getFolderPath(pathname: String): String {

        if (!pathname.contains("/") && pathname.contains(".")) {
            return "./"
        }

        try {
            val lastIndexOf = pathname.lastIndexOf("/")
            return pathname.substring(0, lastIndexOf)
        } catch (e: IndexOutOfBoundsException) {
            return pathname
        }

    }
}
